unit/test_rules/test_rules.c

```diff
@@ -42,6 +42,16 @@
 
 static TestOpt test_opt;
 
+/* Copy internal struct so G_CAST JailRules -> JailRulesPriv can be done. */
+typedef struct jail_rules_priv {
+    char *org_name;
+    char *app_name;
+    JailRules rules;
+    JailDBus dbus_user;
+    JailDBus dbus_system;
+    gint ref_count;
+} JailRulesPriv;
```

Comment:
Maybe move `JailRulesPriv` from `jail_rules.c` to `jail_rules_p.h` and just include header? Because at the next time when `JailRulesPriv` implementation changes, this thing might stay unnoticed.

Reply:
Other than this looks good though.

Reply:
Also don't think it's a good idea to copy-paste such interface stuff...

Reply:
I agree completely with that, thanks for pointing it out. Moved that to `jail_rules_p.h` with adffc9bf7250dd827d8db278fd2b0fe62db96cdf